django/db/models/fields/fetching.py

```diff
@@ -0,0 +1,64 @@
+from contextlib import contextmanager
+
+from asgiref.local import Local
+
+
+def FETCH_ONE(fetcher, instance, **kwargs):
+    fetcher.fetch((instance,))
+
+
+def FETCH_PEERS(fetcher, instance, **kwargs):
+    if instance._state.peers:
+        instances = [
+            peer
+            for weakref_peer in instance._state.peers
+            if (peer := weakref_peer()) is not None
+        ]
+    else:
+        # Peers aren’t tracked for QuerySets returning a single instance
+        instances = (instance,)
+
+    fetcher.fetch(instances)
+
+
+class LazyFieldAccess(Exception):
+    """Blocked lazy access of a model field."""
+
+    pass
+
+
+def RAISE(fetcher, instance, **kwargs):
```

Comment:
Should we get rid of the `**kwargs`? Were they meant to be used for something?

Reply:
I left them to leave space for future extensions, if users wrap the calls. But I guess we don’t have that for the callables in `django.db.models.deletion`, so we can drop them.

Reply:
I'm not that familiar with the back compat policy but the kwargs feels unnecessary
if I had a custom fetch mode I would understand if a feature release broke it due to signature change and it would break loud and fast